cmd/sync-tags/main.go

```diff
@@ -258,71 +279,69 @@ func main() {
 				if _, err2 := os.Stat("Godeps/Godeps.json"); err2 == nil {
 					fmt.Printf("Updating Godeps.json to point to %s tag.\n", bName)
 					changed, err = updateGodepsJsonWithTaggedDependencies(bName, dependentRepos)
 					if err != nil {
 						glog.Fatalf("Failed to update Godeps.json for tag %s: %v", bName, err)
 					}
 				}
-			} else if err == nil {
-				fmt.Printf("Updating go.mod and go.sum to point to %s tag.\n", bName)
-				changed, err = updateGomodWithTaggedDependencies(bName, dependentRepos)
-				if err != nil {
-					glog.Fatalf("Failed to update go.mod and go.sum for tag %s: %v", bName, err)
-				}
-				goModChanged = true
-			}
-
-			if goModChanged && *generateGodeps {
-				fmt.Printf("Regenerating Godeps.json from go.mod.\n")
-				if err := regenerateGodepsFromGoMod(); err != nil {
-					glog.Fatalf("Failed to regenerate Godeps.json from go.mod: %v", err)
+			} else {
+				if publishSemverTag {
+					changed = updateGoModAndGodeps(semverTag, dependentRepos, *generateGodeps, true)
+				} else {
+					changed = updateGoModAndGodeps(bName, dependentRepos, *generateGodeps, false)
 				}
 			}
 
 			if changed {
-				fmt.Printf("Adding extra commit fixing dependencies to point to %s tags.\n", bName)
-				publishingBotNow := publishingBot
-				publishingBotNow.When = time.Now()
-				bh, err = wt.Commit(fmt.Sprintf("Fix dependencies to point to %s tag", bName), &gogit.CommitOptions{
-					All:       true,
-					Author:    &publishingBotNow,
-					Committer: &publishingBotNow,
-				})
-				if err != nil {
-					glog.Fatalf("Failed to commit changes to fix dependencies to point to %s tag: %v", bName, err)
+				if publishSemverTag {
+					bh = createCommitToFixDeps(wt, semverTag)
+				} else {
+					bh = createCommitToFixDeps(wt, bName)
 				}
 			}
 		}
 
-		// create prefixed annotated tag
-		fmt.Printf("Tagging %v as %q.\n", bh, bName)
-		err = createAnnotatedTag(bh, bName, tag.Tagger.When, dedent.Dedent(fmt.Sprintf(`
+		// create semver annotated tag
+		if publishSemverTag && !semverTagExists {
+			fmt.Printf("Tagging %v as %q.\n", bh, semverTag)
+			err = createAnnotatedTag(bh, semverTag, tag.Tagger.When, dedent.Dedent(fmt.Sprintf(`
 			Kubernetes release %s
 
 			Based on https://github.com/kubernetes/kubernetes/releases/tag/%s
 			`, name, name)))
-		if err != nil {
-			glog.Fatalf("Failed to create tag %q: %v", bName, err)
+			if err != nil {
+				glog.Fatalf("Failed to create tag %q: %v", semverTag, err)
+			}
+			createdTags = append(createdTags, semverTag)
+		}
+
+		// create non-semver prefixed annotated tag
+		if !nonSemverTagExists {
+			fmt.Printf("Tagging %v as %q.\n", bh, bName)
+			err = createAnnotatedTag(bh, bName, tag.Tagger.When, dedent.Dedent(fmt.Sprintf(`
+				Kubernetes release %s
+
+				Based on https://github.com/kubernetes/kubernetes/releases/tag/%s
+				`, name, name)))
+			if err != nil {
+				glog.Fatalf("Failed to create tag %q: %v", bName, err)
+			}
+			createdTags = append(createdTags, bName)
```

Comment:
all this look idem-potent and independent. What confuses me is that the commits these tags (semver and non-semver) use, are the same. I would expect different ones, pointing to the respective tags of the dependencies.

Reply:
> What confuses me is that the commits these tags (semver and non-semver) use, are the same. I would expect different ones, pointing to the respective tags of the dependencies.

That was the initial approach, but that exposed us to the pseudoversion problem. Having them point to the `v0.x.y` is not ideal, but that solves the pseudoversion problem for us.

Also, having them point to different commits would mean that `kubernetes-1.x.y` and `v0.x.y` won't be equivalent (in terms of sha) anymore. Prior discussion: https://github.com/kubernetes/publishing-bot/pull/210#discussion_r354098096

Reply:
> What confuses me is that the commits these tags (semver and non-semver) use, are the same. I would expect different ones, pointing to the respective tags of the dependencies.

There is little benefit to shipping a non-semver tag with a go.mod pointing to other non-semver tags, since the tags are tidied to v0.0.0-... before commit, making the go.mod files almost unusable.